File: Data Structures & Algorithms/two-integer-sum/submission-2.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        // target = 7
        // nums = [3, 4, 5, 6]
        // {3: 0, 3:}

        //nums[i] + nums[j] == target
        //i != j (i * j must be different)
        Map<Integer, Integer> map = new HashMap<>();

        for(int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if(map.containsKey(complement)) {
                return new int[]{map.get(complement), i};
            } 
            map.put(nums[i], i);
        }

        return new int[]{-1, -1};
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-2.cs
public class Solution {
    public bool hasDuplicate(int[] nums) {
        Dictionary<int, int> map = new Dictionary<int, int>();
        for(int i = 0; i < nums.Length; i++) {
            if(map.ContainsKey(nums[i])) return true;
            map.Add(nums[i], i);
        }
        return false;
    }
}